### PR TITLE
feat: consider-code-local-engine

### PR DESCRIPTION
**Pull request by shirLupo**
As part of [NEBULA-1003](https://snyksec.atlassian.net/browse/NEBULA-1003)
This PR set the broker support snyk code local engine
We have added the need from local code engine of GIT_CLIENT_URL

also code agent and snyk code local engine cannot be supported in parallel, only one of them

[NEBULA-1003]: https://snyksec.atlassian.net/browse/NEBULA-1003?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: charts/snyk-broker/templates/broker_deployment.yaml [hunk 1]
```diff
@@ -264,7 +264,11 @@ spec:
          {{- if .Values.enableCodeAgent }}
             - name: GIT_CLIENT_URL
               value: {{ default (printf "http://code-agent-service:%s" (.Values.deployment.container.caSnykPort | toString)) .Values.gitClientUrl }}
-         {{- end }}     
+         {{- end }}
+         {{- if .Values.enableCodeLocalEngine }}
```
Comment (metju90) [inline charts/snyk-broker/templates/broker_deployment.yaml:268]: I would use `else if` here, so we prevent the case where we declared `GIT_CLIENT_URL` twice.

File: charts/snyk-broker/templates/broker_deployment.yaml [hunk 2]
```diff
@@ -312,8 +316,10 @@ spec:
          {{ else }}
           {{- if or (eq .Values.scmType "github-com") (eq .Values.scmType "github-enterprise") (eq .Values.scmType "bitbucket-server") (eq .Values.scmType "gitlab") (eq .Values.scmType "azure-repos") }}
          # Default Values to allow Snyk Code Snippets and Snyk IaC
+            {{- if not .Values.enableCodeLocalEngine}}
```
Comment (metju90) [inline charts/snyk-broker/templates/broker_deployment.yaml:319]: Maybe add this condition in line 317?

File: charts/snyk-broker/templates/broker_deployment.yaml [hunk 3]
```diff
@@ -264,7 +264,11 @@ spec:
          {{- if .Values.enableCodeAgent }}
             - name: GIT_CLIENT_URL
               value: {{ default (printf "http://code-agent-service:%s" (.Values.deployment.container.caSnykPort | toString)) .Values.gitClientUrl }}
-         {{- end }}     
+         {{- end }}
+         {{- if .Values.enableCodeLocalEngine }}
+            - name: GIT_CLIENT_URL
+              value: {{ .Values.gitClientUrl }}
```
Comment (metju90) [inline charts/snyk-broker/templates/broker_deployment.yaml:270]: From the values, we will pass `"http://{{ .Release.Name}}-foo-bar"`. Can you use `{{ tpl .Values.gitClientUrl .}}` here?

File: charts/snyk-broker/values.yaml [hunk 4]
```diff
@@ -148,6 +150,10 @@ caImage: "latest"
 # Only adjust this value if advised to by Snyk Representative. This is the URL of the Snyk Code Agent. This helm chart already connects everything. 
 gitClientUrl: ""
 
+##### Code Local Engine #####
+
+# Set to 'true' to enable Code Local Engine
+enableCodeLocalEngine: ""
```
Comment (metju90) [inline charts/snyk-broker/values.yaml:156]: Can we use the full name? `enableSnykCodeLocalEngine`

File: charts/snyk-broker/values.yaml [hunk 5]
```diff
@@ -136,9 +136,11 @@ crImage: "latest"
 
 # Set to 'true' to enable Code Agent
 enableCodeAgent: ""
+
 # Only adjust this value if advised to by Snyk Representative. Used to upload content to non-standard environment. 
 upstreamUrlCodeAgent: ""
 
+
```
Comment (metju90) [inline charts/snyk-broker/values.yaml:143]: Unnecessary new lines

File: charts/snyk-broker/templates/broker_deployment.yaml (hunk 2 above)
Comment (shirlupo) [reply charts/snyk-broker/templates/broker_deployment.yaml:319]: I thought of it but it 371 also affecting IaC and I do not want to touch it, what do you think?

Comment (metju90) [reply charts/snyk-broker/templates/broker_deployment.yaml:319]: You are right!

**Review by metju90 — APPROVED** [review]
LGTM

**Review by aarlaud — APPROVED** [review]

**Comment by pkey** [comment]
@shirlupo sweet!

**Pull request merged**